src/algebra/opposites.lean

```diff
@@ -253,6 +253,26 @@ instance _root_.monoid.to_opposite_mul_action [monoid α] : mul_action (opposite
   one_smul := mul_one,
   mul_smul := λ x y r, (mul_assoc _ _ _).symm }
 
+instance _root_.is_scalar_tower.opposite_mid {M N} [monoid N] [has_scalar M N]
+  [smul_comm_class M N N] :
+  is_scalar_tower M Nᵒᵖ N :=
+⟨λ x y z, mul_smul_comm _ _ _⟩
+
+instance _root_.smul_comm_class.opposite_mid {M N} [monoid N] [has_scalar M N]
+  [is_scalar_tower M N N] :
+  smul_comm_class M Nᵒᵖ N :=
+⟨λ x y z, by { induction y using opposite.op_induction, simp [smul_mul_assoc] }⟩
+
+instance _root_.is_scalar_tower.opposite_right {M N} [has_scalar M N] [has_scalar M α]
+  [has_scalar N α] [is_scalar_tower M N α] :
+  is_scalar_tower M N αᵒᵖ :=
+⟨λ x y z, unop_injective $ smul_assoc _ _ _⟩
+
+instance _root_.smul_comm_class.opposite_right {M N} [has_scalar M α] [has_scalar N α]
+  [smul_comm_class M N α] :
+  smul_comm_class M N αᵒᵖ :=
+⟨λ x y z, unop_injective $ smul_comm _ _ _⟩
```

Comment:
I think these two instances belong up on line 216, as they about the action there and not the action on line 240

Reply:
(and should be called `opposite.smul_comm_class` etc to match `opposite.module`)